lib/http/2/compressor.rb

```diff
@@ -134,9 +130,9 @@ def dup
       # @return [Array] +[key, value]+
       def dereference(index)
         # NOTE: index is zero-based in this module.
-        STATIC_TABLE[index] or
-          @table[index - STATIC_TABLE.size] or
-          raise CompressionError.new("Index too large")
+        STATIC_TABLE[index] ||
```

Comment:
This particular point is what I disagree with https://github.com/bbatsov/ruby-style-guide .
I prefer `and`/`or` over `&&`/`||` when its all about flow control. `&&`/`||` should convey value, not control IMHO.
- condition `and` action
- condition `&&` condition
I prefer `or` here.  However, rewriting with nested if with condition `||` condition would be better in this case.

I tend to mix `&&` and `and` in a single sentence and it's a bad habit. I should avoid it.
- condition `||` condition `and` action
This should be fixed.